data/binding/preference.go

```diff
@@ -0,0 +1,133 @@
+// auto-generated
+// **** THIS FILE IS AUTO-GENERATED, PLEASE DO NOT EDIT IT **** //
+
+package binding
+
+import "fyne.io/fyne"
+
+// Because there is no preference listener yet we connect any listeners asking for the same key.
+var prefBinds = make(map[string]DataItem)
+
+type prefBoundBool struct {
+	base
+	key string
+	p   fyne.Preferences
+}
+
+// BindPreferenceBool returns a bindable bool value that is managed by the application preferences.
+// Changes to this value will be saved to application storage and when the app starts the previous values will be read.
+func BindPreferenceBool(key string, p fyne.Preferences) Bool {
+	if listen, ok := prefBinds[key]; ok {
+		if l, ok := listen.(Bool); ok {
+			return l
+		}
+		fyne.LogError("A previous preference binding exists with different type for key: "+key, nil)
```

Comment:
Could this be extracted to a helper `logKeyTypeError` defined at the top of this file?